File: 백태균/week06/68644_두_개_뽑아서_더하기.java
import java.io.*;
import java.util.*;

class Solution {
    public int[] solution(int[] numbers) {
        // List<Integer> list = new ArrayList<>();
        Set<Integer> set = new HashSet<>();  // Set 사용

        for (int i=0; i<numbers.length-1; i++) {
            int a = numbers[i];

            for (int j=i+1; j<numbers.length; j++) {
                int b = numbers[j];
                int sum = a + b;

                // if (!list.contains(sum)) {
                //     list.add(sum);
                // }

                set.add(sum);
            }
        }

        // int[] answer = new int[list.size()];
        // for (int i=0; i<list.size(); i++) {
        //     answer[i] = list.get(i);
        // }

        int[] answer = new int[set.size()];
        int idx = 0;

        for (int num : set) {
            answer[idx] = num;
            idx++;
        }

        Arrays.sort(answer);

        return answer;
    }
}

File: 김진효/week06/68644_두_개_뽑아서_더하기.py
# [성능 요약] 메모리: 11.6 MB 시간: 1.97 ms 

def solution(numbers):
    answer = []

    l = len(numbers)

    for i in range(l):
        for j in range(i+1, l):
            s = numbers[i] + numbers[j]
            if s not in answer:
                answer.append(s)

    return sorted(answer)

print(solution([2,1,3,4,1]))